eo-maven-plugin/src/main/java/org/eolang/maven/hash/ChResolve.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016-2022 Objectionary.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven.hash;
+
+import java.nio.file.Path;
+
+/**
+ * Resolve {@link CommitHash} depending on the
+ * parameters.
+ *
+ * @since 0.28.14
+ */
+public final class ChResolve implements CommitHash {
```

Comment:
@MikhailLipanin The object name `ChResolve` almost doesn't tell us about its function. Maybe we might rename it, what do you think?

Reply:
@volodya-lombrozo, This object resolves, which one `CommitHash` to use, so this is because it has such name. Do you have  suggestions about another name?

Reply:
@MikhailLipanin 
> A much more sophisticated and time-consuming way of naming classes is by the [DDD paradigm](https://en.wikipedia.org/wiki/Domain-driven_design), which suggests using nouns after entities in the real world, like port, car, book, story, user, socket, and so on. Identifying the right entities is a big challenge for a software architect. A bad one would just resort to ConnectionFactoryUtils or DefaultListableBeanFactory and call it a day. A more professional one may spend hours or days, but will eventually come up with something more domain-driven.

Please, read that article https://www.yegor256.com/2020/03/03/prefixed-naming.html

It's not directly about "nouns", but the key idea that it's better to name closes like `Ch<Adjective>` or `Ch<Noun>`.
"Resolve" - it's a verb.

Reply:
@volodya-lombrozo which one is better: `ChResolving` or `ChResolved`?

Reply:
@MikhailLipanin Non of them. What do you "Resolve"? Moreover they both are participle. If you can't come up with the name, you can leave one more puzzle to avoid block.

Reply:
added todo